### PR TITLE
[pdata] Rename Map.Upsert* methods to Map.Put*

### PR DESCRIPTION
**Pull request by dmitryax**
**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector/issues/6059

### CODE REVIEW

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Will wait until after the SIG meeting to merge.

**Review by codeboten — APPROVED** [review]
~Will have to merge with the failing contrib check~

@dmitryax's pr https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14118 solves this 👍🏻

**Review by mx-psi — APPROVED** [review]

**Pull request merged**
